File: FacturaTest.java
import java.util.Scanner;

public class FacturaTest {
    public static void main(String args[]){
        Factura ferre=new Factura();
        Scanner input=new Scanner(System.in);
        
        System.out.println("Ingresa el  número de pieza: ");
        ferre.numPieza=input.nextLine();
        
        System.out.println("Agrega la descripción: ");
        ferre.descripcion=input.nextLine();
        
        System.out.println("Cantidad: ");
        ferre.cantidad=input.nextInt();
        
        System.out.println("Precio: ");
        ferre.precio=input.nextDouble();
        
        if (ferre.precio<0) ferre.precio=0.0;
        if (ferre.cantidad<0) ferre.cantidad=0;
        
        System.out.println("Factura: "+ferre.obtenerMontoFactura());
    }
}

File: FacturaTest.cs
using System;

class FacturaTest
{
    static void Main()
    {
        Factura ferre = new Factura();

        Console.WriteLine("Ingresa el número de pieza: ");
        ferre.numPieza = Console.ReadLine();

        Console.WriteLine("Agrega la descripción: ");
        ferre.descripcion = Console.ReadLine();

        Console.WriteLine("Cantidad: ");
        ferre.cantidad = int.Parse(Console.ReadLine());

        Console.WriteLine("Precio: ");
        ferre.precio = double.Parse(Console.ReadLine());

        if (ferre.precio < 0)
            ferre.precio = 0.0;

        if (ferre.cantidad < 0)
            ferre.cantidad = 0;

        Console.WriteLine("Factura: " + ferre.obtenerMontoFactura());
    }
}